### PR TITLE
Add end-to-end tests for Azure classic and Azure native

### PR DESCRIPTION
**Pull request by myhau**
## Task

Resolves: https://github.com/VirtuslabRnD/pulumi-kotlin/issues/116
Resolves: https://github.com/VirtuslabRnD/pulumi-kotlin/issues/115

## Description

todo

## To do

- [x] pass subscription-id and tenant-id as env variables here

```
fun getVirtualMachine(virtualMachineId: String): VirtualMachine {
    // TODO: parametrize?
    val subscriptionId = "3565bcd1-dd9b-45a8-9d7c-8880aaaa9a9f"
    val tenantId = "1ff3e4bd-41d1-4df2-b776-8d3fcbf77184"
```

### CODE REVIEW

File: .github/workflows/e2e_tests.yml [hunk 1]
```diff
@@ -11,6 +11,84 @@ permissions:
...
+        uses: azure/login@v1
+        id: auth
+        with:
+          # These values were copied from jvm-lab/identity-federation stack's output
+          # The Azure tenant/subscription that is used here belongs to @myhau (Michal Fudala)
+          tenant-id: '1ff3e4bd-41d1-4df2-b776-8d3fcbf77184' # tenant-id output of jvm-lab/identity-federation/dev stack
+          subscription-id: '3565bcd1-dd9b-45a8-9d7c-8880aaaa9a9f' # subscription-id output of jvm-lab/identity-federation/dev stack
+          client-id: '4169b156-839a-48c2-846f-d7c079331c71' # service-principal-client-id output of jvm-lab/identity-federation/dev stack
+      - name: Publish Pulumi Kotlin Azure to Maven Local
+        run: ./gradlew publishPulumiAzureE2ePublicationToMavenLocal -x formatKotlinPulumiAzure -x dokkaPulumiAzureJavadoc
```
Comment (jplewa) [inline .github/workflows/e2e_tests.yml:37]: Remove ` -x formatKotlinPulumiAzure -x dokkaPulumiAzureJavadoc` (they don't get triggered with the E2E task)

File: .github/workflows/e2e_tests.yml [hunk 2]
```diff
@@ -11,6 +11,84 @@ permissions:
...
+      - name: Publish Pulumi Kotlin Random to Maven Local
+        run: ./gradlew publishPulumiRandomE2ePublicationToMavenLocal -x formatKotlinPulumiRandom -x dokkaPulumiRandomJavadoc
```
Comment (jplewa) [inline .github/workflows/e2e_tests.yml:39]: Remove `-x formatKotlinPulumiRandom -x dokkaPulumiRandomJavadoc` (they don't get triggered with the E2E task)

File: .github/workflows/e2e_tests.yml [hunk 3]
```diff
@@ -11,6 +11,84 @@ permissions:
...
+        uses: azure/login@v1
+        id: auth
+        with:
+          # These values were copied from jvm-lab/identity-federation stack's output
+          # The Azure tenant/subscription that is used here belongs to @myhau (Michal Fudala)
+          tenant-id: '1ff3e4bd-41d1-4df2-b776-8d3fcbf77184' # tenant-id output of jvm-lab/identity-federation/dev stack
+          subscription-id: '3565bcd1-dd9b-45a8-9d7c-8880aaaa9a9f' # subscription-id output of jvm-lab/identity-federation/dev stack
+          client-id: '4169b156-839a-48c2-846f-d7c079331c71' # service-principal-client-id output of jvm-lab/identity-federation/dev stack
+      - name: Publish Pulumi Kotlin Azure Native to Maven Local
+        run: ./gradlew publishPulumiAzure-nativeE2ePublicationToMavenLocal -x formatKotlinPulumiAzure-native -x dokkaPulumiAzure-nativeJavadoc
```
Comment (jplewa) [inline .github/workflows/e2e_tests.yml:76]: https://github.com/VirtuslabRnD/pulumi-kotlin/pull/155/files#r1049691390

File: .github/workflows/e2e_tests.yml [hunk 4]
```diff
@@ -11,6 +11,84 @@ permissions:
...
+      - name: Publish Pulumi Kotlin Random to Maven Local
+        run: ./gradlew publishPulumiRandomE2ePublicationToMavenLocal -x formatKotlinPulumiRandom -x dokkaPulumiRandomJavadoc
```
Comment (jplewa) [inline .github/workflows/e2e_tests.yml:78]: https://github.com/VirtuslabRnD/pulumi-kotlin/pull/155/files#r1049691390

File: .github/workflows/e2e_tests.yml [hunk 5]
```diff
@@ -11,6 +11,84 @@ permissions:
...
+    name: Run end-to-end tests of Pulumi Kotlin Azure Native
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      id-token: write
+    steps:
+      - name: Check out project sources
+        uses: actions/checkout@v2
+      - name: Set up Gradle
+        uses: gradle/gradle-build-action@v2
```
Comment (jplewa) [inline .github/workflows/e2e_tests.yml:63]: This step can be removed

File: src/e2e/kotlin/com/virtuslab/pulumikotlin/azure/AzureNativeE2eTest.kt [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+        val virtualMachine = getVirtualMachine(pulumi.getStackOutput<PulumiStackOutput>().virtualMachineId)
+
+        assertVmExists(virtualMachine)
+    }
+
+    @AfterTest
+    fun cleanupTest() {
+//        pulumi.destroy()
+//        pulumi.rmStack()
```
Comment (jplewa) [inline src/e2e/kotlin/com/virtuslab/pulumikotlin/azure/AzureNativeE2eTest.kt:33]: Uncomment

File: examples/gcp-sample-project/Pulumi.example.yaml [hunk 7]
```diff
@@ -1,2 +1,3 @@
 config:
   gcp:project: jvm-lab
+
```
Comment (jplewa) [inline examples/gcp-sample-project/Pulumi.example.yaml:3]: (super minor) I'd revert this change, since it doesn't really contribute anything :p

File: src/main/kotlin/com/virtuslab/pulumikotlin/scripts/ComputeSchemaSubsetScript.kt [hunk 8]
```diff
@@ -57,12 +60,24 @@ private const val HELP =
...
-    private val context: Context by option().enum<ExposedContext>().convert { it.toContext() }.required()
-    private val name: String by option().required()
+    private val existingSchema: String? by option()
+    private val nameAndContexts: List<NameWithContext>
+        by option("--name-and-context")
+            .transformValues(2) { NameWithContext(it[0], ExposedContext.valueOf(it[1].capitalize()).toContext()) }
+            .multiple()
+            .help(
+                "Specify multiple names and their context (type / function). " +
+                    "For example --name-and-context VirtualMachine --name-and-context",
```
Comment (jplewa) [inline src/main/kotlin/com/virtuslab/pulumikotlin/scripts/ComputeSchemaSubsetScript.kt:78]: Is this example correct? There's no context here and the second `--name-and-context` isn't followed by anything

**Review by jplewa — APPROVED** [review]

File: src/main/kotlin/com/virtuslab/pulumikotlin/scripts/ComputeSchemaSubsetScript.kt (hunk 8 above)
Comment (myhau) [reply src/main/kotlin/com/virtuslab/pulumikotlin/scripts/ComputeSchemaSubsetScript.kt]: done, also updated other docs

**Review by ddzikon — APPROVED** [review]
